R/utils-shared.R

```diff
@@ -226,12 +226,27 @@ splitDataGaps <- function(data, ts){
   
   data_list <- data[[ts$field[1]]]
   dataSplit <- list()
+  
+  hasGaps <- "gaps"  %in% names(data_list) && !isEmptyOrBlank(data_list$gaps)
+  hasEstimatedRangesAsGaps <- !isEmptyOrBlank(ts$estimated) && !ts$estimated == FALSE && 
+    "estimatedPeriods"  %in% names(data_list) && 
+    !isEmptyOrBlank(data_list$estimatedPeriods)
+  
+  if(hasGaps || hasEstimatedRangesAsGaps){
 
-  if("gaps"  %in% names(data_list) && !isEmptyOrBlank(data_list$gaps)){
-
-    #might need formatDates instead?
-    startGaps <- flexibleTimeParse(data_list$gaps$startTime, timezone = data$reportMetadata$timezone)
-    endGaps <- flexibleTimeParse(data_list$gaps$endTime, timezone = data$reportMetadata$timezone)
+    if(hasGaps) {
+      #might need formatDates instead?
+      startGaps <- flexibleTimeParse(data_list$gaps$startTime, timezone = data$reportMetadata$timezone)
+      endGaps <- flexibleTimeParse(data_list$gaps$endTime, timezone = data$reportMetadata$timezone)
+    }
+    
+    if(hasEstimatedRangesAsGaps) {
+      startEstimated <- flexibleTimeParse(data_list$estimatedPeriods$startDate, timezone = data$reportMetadata$timezone)
+      endEstimated <- flexibleTimeParse(data_list$estimatedPeriods$endDate, timezone = data$reportMetadata$timezone)
+      
+      #TODO append start/end times to startGaps and endGaps
```

Comment:
@lindsaycarr this TODO still needs to be done. I stopped short of completing this here. It looks like the date ranges are of different formats. Also I think some logic needs to be added in the case their are no specified gaps but are estimated ranges that need to be used as gaps. I think once the TODO is complete the code will work as you had it before.